.circleci/config.yml

```diff
@@ -195,7 +195,7 @@ jobs:
           paths:
             - node_modules
             - .yarn/cache
-      - run: yarn eslint src
+      - run: yarn eslint --max-warnings=0 src
```

Comment:
@zarsky-broad is this the place we could append `integration-test/src` so test files are linted as well?

```suggestion
      - run: yarn eslint --max-warnings=0 src integration-test/src
```

Reply:
If we're adding to the files checked by ESLint, do we want to stop at `integration-tests/src`? Or check all JS files in the project?

Reply:
There was concern about things taking a long time if we are linting dependency libraries. I would say lint as much of our own code as you can without introducing speed issues.

Reply:
(or we could be grownups and add an .eslintignore file for node modules and the yarn cache)

Reply:
I tried a few runs on my machine. With node_modules and minified JS files ignored, running ESLint on the entire project vs src added ~0.5s to its ~7s second runtime.

Reply:
FWIW I'd be happy to take that penalty 😊

Reply:
Great. I'll go ahead and merge this PR (to reduce the chance of additional warnings getting committed to dev while this stays open) and open another one to check everything with ESLint.

Reply:
Opened #2873.